### ISSUE TITLE
There is no need to return a long in TupleXByteBuf.calculateRelativeIndex

### ISSUE THREAD

**Issue by Liyixin95** [rsocket/rsocket-java#692]
The values right of the equal are `int` ,so the result will be `int`  if there wasn't any cast.Just like the code below :  
> `relativeIndex = oneReadIndex + index;`

The `oneReadIndex ` and `index` are both `int`, so the `relativeIndex ` will be `int` too,but `relativeIndex ` was defined as a `long` and the return value was a `long `too.The right code would be : 

> `relativeIndex = (long)oneReadIndex + index;`

But I read all the call of this method.Both of them cast the return value to int.And the TupleXBytebuf are only used to combine `HeadFrame`,`MetaDataFrame `and `PlayloadFrame`
,according to the protocol,the total length of these three frame are not large then `int`,so I think return a int is enough.

In conclusion,there is too way to fix the code,add a cast or change the return value, and I would like to provide a PR.
